_blogposts/compiler/a-story-of-lazy-encoding.mdx

```diff
@@ -0,0 +1,122 @@
+---
+author: hongbo
+date: "2020-05-15 18:00" 
+previewImg:
+category: compiler
+title: New Lazy Encoding in BuckleScript
+description: |
+  Highlights of our newest changes to the internal representation of exceptions
+  and how it will provide better stacktraces to our users.
+canonical: https://bucklescript.github.io/blog/2020/05/06/exception-encoding
+---
+
+---
+title: Enhanced lazy encoding in BuckleScript
+---
+
+
+
+Recently we made some significant improvements with our new encoding for lazy values, and we find it so exciting that we want to highlight the changes. The new encoding generates very idiomatic JS output like hand-written code.
+
+For people who are not familiar with lazy evaluation, it is documented here: https://ocaml.org/releases/4.10/htmlman/expr.html#sss:expr-lazy.
+
+# What's the difference?
```

Comment:
h1 headers are reserved for the title, the content does start with a level 2 header

```suggestion
## Comparison between the old and new lazy encoding
```